src/writer/serve.py

```diff
@@ -546,10 +547,36 @@ async def _handle_incoming_edit_message(websocket: WebSocket, session_id: str, r
                 ))
         elif req_message.type == "codeSaveRequest":
             app_runner.save_code(
-                session_id, req_message.payload["code"])
+                session_id, req_message.payload["code"], req_message.payload["path"])
         elif req_message.type == "codeUpdate":
-            app_runner.update_code(
-                session_id, req_message.payload["code"])
+            app_runner.update_code(session_id, req_message.payload["code"])
+        elif req_message.type == "loadSourceFile":
+            path = os.path.join(*req_message.payload['path'])
+            try:
+                response.payload = { "content": app_runner.load_persisted_script(path) }
+            except FileNotFoundError as error:
+                logging.warning(f"could not load script at {path}", error)
+                response.payload = {"error": str(error)}
```

Comment:
maybe we have to exten the `mail protocol` to get it on all the websocket message